src/seedu/addressbook/Main.java

```diff
@@ -106,11 +105,14 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
-        try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+        try {
```

Comment:
Perhaps you can include `.setData(..)` and `.execute()` in the try block, as `RuntimeException` may be caused by these operations.